client/actions/editActions.js

```diff
@@ -95,8 +114,8 @@ export const setSection = (sectionIndex) => ({
   }
 })
 
-export const newSection = (type, sectionIndex) => {
-  const section = setupSection(type)
+export const newSection = (type, sectionIndex, attrs = {}) => {
+  const section = extend(setupSection(type), attrs)
```

Comment:
Totally a matter of preference, but since `extend` is a shallow can use ES7 spread: 
```javascript
const section {...setupSection(type), ...attrs}
```
Or built in ES6 Object.assign
```javascript
const section = Object.assign(setupSection(type), attrs)
```